lib/r10k/features.rb

```diff
@@ -0,0 +1,5 @@
+require 'r10k/feature/collection'
+
+module R10K
+  Features = R10K::Feature::Collection.new
```

Comment:
In the long term we need a better way to handle singletons like this, but I don't think that now is the time to implement something more complex. Is this sufficent?

 For reference, this would be used like so:

```ruby
# Adding features

R10K::Features.add(:rugged, :libraries => 'rugged')

R10K::Features.add(:shellgit) do
  !!R10K::Util::Commands.which('git')
end

# Checking for features

R10K::Features.available?(:rugged)
```